### PR TITLE
Feature/[LRA-7] Delete Product

### PR DESCRIPTION
**Pull request by Shalin21**
- Added remove service logic
- Added new action types for remove
- Added remove logic to product reducer
- Added remove logic to product saga
- Refactored product saga with new function names
- Added page folder for ProductDetails page

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
 import './App.css';
-import ProductDetails from './Pages/ProductDetails';
+import ProductDetails from './Pages/Product/ProductDetails/ProductDetails';
```
Comment (mkozachok) [inline src/App.tsx]: it is so "long import",  propose 2 option:
1. Move to `ProductDetails` to  `./components/ProductDetails` and export it from `./components/index.ts`
2. Move to `ProductDetails` to  `./Pages/Product/components` and export it from `./components/index.ts`

I like 1 option. What you think?

File: src/Pages/Product/ProductDetails/ProductDetails.tsx [hunk 2]
```diff
@@ -13,11 +12,17 @@ const ProductDetails = function({ match }: RouteComponentProps<TParams>) {
   const dispatch = useDispatch();
   const product = useSelector((state: IProductState) => state.productReducer.currentProduct);
   const isLoading = useSelector((state: IProductState) => state.productReducer.isLoading);
-  
+
+  const removeProduct = function (): any {
```
Comment (mkozachok) [inline src/Pages/Product/ProductDetails/ProductDetails.tsx]: this function return `undefined`

File: src/Pages/Product/ProductDetails/ProductDetails.tsx [hunk 3]
```diff
@@ -13,11 +12,17 @@ const ProductDetails = function({ match }: RouteComponentProps<TParams>) {
...
+    dispatch(removeProductByIdStart(match.params.product_id));
+  }
+
   useEffect(() => {
     dispatch(getProductByIdStart(match.params.product_id));
   }, [dispatch, match.params.product_id]);
 
+  
+
```
Comment (mkozachok) [inline src/Pages/Product/ProductDetails/ProductDetails.tsx]: Please remove extra space

File: src/store/reducers/productReducer.ts [hunk 4]
```diff
@@ -23,6 +23,21 @@ const productReducer = (state = initState, action: IProductAction) => {
...
+        isLoading: true,
+      }
+    case productActionTypes.REMOVE_PRODUCT_BY_ID__SUCCESS:
+      return {
+        currentProduct: null,
+        isLoading: false,
+      }
+    case productActionTypes.REMOVE_PRODUCT_BY_ID__FAILURE:
+      return {
+        currentProduct: null,
```
Comment (mkozachok) [inline src/store/reducers/productReducer.ts]: On error, you do not need to clear state. 
`{
...state, isLoading: false,
}`

File: src/store/sagas/productSaga.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ function* getProductByIdSaga(action: any) {
     const currentProduct = yield call(productAPI.getProductByID, action.product_id);
     // comment the line above and uncomment lines below if current endpoint for the API doesn't work
```
Comment (mkozachok) [inline src/store/sagas/productSaga.ts]: Please remove comments

File: src/App.tsx (hunk 1 above)
Comment (Shalin21) [reply src/App.tsx]: Fixed with 1st one. Also refactored old components import/export to index.ts in /Components forlder

File: src/Pages/Product/ProductDetails/ProductDetails.tsx (hunk 2 above)
Comment (Shalin21) [reply src/Pages/Product/ProductDetails/ProductDetails.tsx]: Changed return type to void

File: src/Pages/Product/ProductDetails/ProductDetails.tsx (hunk 3 above)
Comment (Shalin21) [reply src/Pages/Product/ProductDetails/ProductDetails.tsx]: Fixed

File: src/store/reducers/productReducer.ts (hunk 4 above)
Comment (Shalin21) [reply src/store/reducers/productReducer.ts]: Fixed

File: src/store/sagas/productSaga.ts (hunk 5 above)
Comment (Shalin21) [reply src/store/sagas/productSaga.ts]: Fixed

File: src/App.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
 import './App.css';
-import ProductDetails from './Pages/ProductDetails';
+import { ProductDetails } from './Components';
 import {homePage, loginPage, signUpPage} from "./Pages/";
```
Comment (mkozachok) [inline src/App.tsx]: test

**Review by mkozachok — APPROVED** [review]

File: src/Components/ProductDetails/ProductDetails.tsx [hunk 7]
```diff
@@ -1,19 +1,22 @@
...
+import { getProductByIdStart, removeProductByIdStart } from '../../store/actions/productActions';
+import { IProductState } from '../../types/product';
 import { RouteComponentProps } from 'react-router';
 
 type TParams = {
   product_id: string,
 }
 
-const ProductDetails = function({ match }: RouteComponentProps<TParams>) {
+export const ProductDetails = function({ match }: RouteComponentProps<TParams>) {
```
Comment (KonstantinChabanenko) [inline src/Components/ProductDetails/ProductDetails.tsx]: Пожалуйста, напиши через arrow function.

File: src/Components/ProductDetails/ProductDetails.tsx [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
   const dispatch = useDispatch();
   const product = useSelector((state: IProductState) => state.productReducer.currentProduct);
   const isLoading = useSelector((state: IProductState) => state.productReducer.isLoading);
-  
+
+  const removeProduct = function (): void {
```
Comment (KonstantinChabanenko) [inline src/Components/ProductDetails/ProductDetails.tsx]: Тоже arrow function

File: src/Components/ProductDetails/ProductDetails.tsx [hunk 9]
```diff
@@ -1,19 +1,22 @@
...
+    dispatch(removeProductByIdStart(match.params.product_id));
```
Comment (KonstantinChabanenko) [inline src/Components/ProductDetails/ProductDetails.tsx]: match.params.product_id вынеси в отдельную переменную ( const product_id = match.params.product_id ) и замени match.params.product_id во всех местах на product_id

File: src/Components/ProductDetails/ProductDetails.tsx (hunk 8 above)
Comment (Shalin21) [reply src/Components/ProductDetails/ProductDetails.tsx]: Fixed

File: src/Components/ProductDetails/ProductDetails.tsx (hunk 9 above)
Comment (Shalin21) [reply src/Components/ProductDetails/ProductDetails.tsx]: Fixed

**Pull request merged**